src/cmd/services/m3query/config/config.go

```diff
@@ -94,10 +101,47 @@ type Configuration struct {
 	// Ingest is the ingest server.
 	Ingest *IngestConfiguration `yaml:"ingest"`
 
+	// Carbon is the carbon configuration.
+	Carbon *CarbonConfiguration `yaml:"carbon"`
+
 	// Limits specifies limits on per-query resource usage.
 	Limits LimitsConfiguration `yaml:"limits"`
 }
 
+// CarbonConfiguration returns the carbon configuration.
+func (c *Configuration) CarbonConfiguration() *CarbonConfiguration {
+	if c.Carbon == nil {
+		// Nothing was specified in the configuration, use defaults.
+		return &CarbonConfiguration{
+			Ingestion: &CarbonIngestionConfiguration{
+				Enabled:       &defaultCarbonIngesterEnabled,
+				ListenAddress: defaultCarbonIngesterListenAddress,
+				Timeout:       &defaultCarbonIngesterTimeout,
+			},
+		}
+	}
+
+	if c.Carbon.Ingestion.Enabled == nil {
+		// If they have set any carbon configuration and haven't explicitly marked
+		// the ingestion as disabled, then assume they want it enabled.
+		c.Carbon.Ingestion.Enabled = &defaultCarbonIngesterEnabled
```

Comment:
Perhaps you could just make all these values setters use getters instead?

i.e.
```go
func (c *CarbonIngestionConfiguration) EnabledValue() bool {
  if c.Enabled != nil {
    return *c.Enabled
  }
  return defaultCarbonIngestConfigurationEnabled
}
```

That way there's no way any field could be nil when accessing it it.